packages/Input/src/Input.js

```diff
@@ -27,9 +26,13 @@ const Input = React.forwardRef((props, ref) => {
   } = props;
 
   const isControlled = value !== undefined;
+  const containerProps = extractChildrenProps(children, InputPropsCollector);
 
   const i18n = useI18n();
 
+  const _refInput = React.useRef();
```

Comment:
Feels overkill to have a variable for this,

can just do `const refInput = ref || React.useRef();` on this next line

Reply:
It is possible if you don't care about the `Rules of Hooks`! 
Actually I tried this originally, it worked, and eslint didn't complain until it got to CI.  But it is a violation of the rules – you should not call hooks conditionally.